cypress/plugins/index.js

```diff
@@ -18,5 +18,6 @@
 module.exports = (on, config) => {
   // `on` is used to hook into various events Cypress emits
   // `config` is the resolved Cypress config
-  on('task', require('@cypress/code-coverage/task'));
+  require('@cypress/code-coverage/task')(on, config)
+  return config
```

Comment:
This is following the instructions on https://github.com/cypress-io/code-coverage

Doing this and updating the package versions made this start passing locally for me.